packages/aws-cdk/lib/api/hotswap/ecs-services.ts

```diff
@@ -117,6 +172,12 @@ export async function isHotswappableEcsServiceChange(
           }),
         );
 
+        if (taskDefRevArn !== undefined) {
+          // Make the task definition registered by hotswap INACTIVE immediately.
+          // This is to prevent describeTaskDefinition API from returning a drifted task definition.
```

Comment:
This I don't get, but I like that the comment is here! Can you try explaining when/why this matters?

If the problem is that when you do this method of hotswap multiple times the updates wouldn't apply anymore after the first hotswap, could we circumvent that by using `DescribeTemplateResources` to get the ARN of the TaskDefinition that we based the updates off of?

Reply:
> If the problem is that when you do this method of hotswap multiple times the updates wouldn't apply anymore after the first hotswap

Yes, that is exactly the problem I want to solve. Since we are taking diff between the old and new template, a task definition to apply patches must be the one deployed by the old template.

And I agree with that using `DescribeTemplateResources` is more reliable. (Thanks, I didn't come up with that!) I'll fix this soon :)